### DISCUSSION TITLE
Question about RIS function

### DISCUSSION THREAD

**Discussion by tlstkddn999** [NVlabs/sionna#499] [Q&A]
Hello, 

I am conducting several experiments related to RIS (Reconfigurable Intelligent Surfaces) using map information extracted from Blender.

I have positioned the RIS on a building wall using the coordinates and direction obtained from Blender, setting its orientation and angle accordingly. Then, following the RIS tutorial, I obtained the phase information of the RIS through ris.focusing_lens given the positions of TX, RX, and RIS.

However, when I computed the paths using scene.compute_paths and checked the preview, the TX-RIS-RX path did not appear. Additionally, when I compared the 'a' and 'tau' values using path.cir(), they were filled with zeros.

In the tutorial, the look_at parameter for the RIS is set to the average position of TX and RX. Doing so positions the RIS in mid-air rather than on the wall of the building.

I am wondering if the path did not appear because the phase obtained from ris.focusing_lens with the RIS attached to the wall (without specifying the look_at parameter as in the tutorial) cannot be realized in practice (i.e., it cannot create the required EM field), thus causing the path to not appear

Additionally, I would like to know if there is a way to visually represent the phase or codebook of the RIS in the preview without having the RX position specified.

Thank you.

<img width="327" alt="output" src="https://github.com/NVlabs/sionna/assets/113810868/f7ca8d06-f4cf-4667-89c9-e4a44881db18">

**Comment by jhoydis** [comment]
Hi @tlstkddn999,

It could be that your RIS is sitting exactly within the wall and might be blocked by it. Could you try moving it a bit out of the wall?
If this does not solve the issue, are you sure that there is a line-of-sight connection between the TX and RIS and also the RIS and RX? 

The phase profile has no impact on whether or not there will be a path. It only impacts the received field.
You can always visualize the phase or amplitude profile via `RiS.phase_profile.show()` or `RiS.amplitude_profile.show()`.
You can assign arbitrary values to the phase/amplitude profiles.  The `phase_gradient_reflector` or `focusing_lens` are just helper functions that you can use if you want; but this is not required.

**Reply by tlstkddn999** [reply]
Thank you. @jhoydis 


Your response helped me resolve the issue.
Upon investigation, it turned out that there was an obstacle between the TX-RIS-RX path, preventing the signal from propagating.
However, I have another question.

When an RIS is created based on the arbitrary location information of the 'RX', the signal will propagate from the RIS to the 'RX location'. In reality, as shown in the attached picture, the signal propagating to the RX location will undergo reflection or diffraction and move in other directions. Is this also accounted for in the simulation, and is it possible to calculate the path in such situations?







<img width="382" alt="image" src="https://github.com/NVlabs/sionna/assets/113810868/9f0f9573-64b6-466b-afdf-8e3ee736d401">
